### DISCUSSION TITLE
Ignore expected differences, such as dates

### DISCUSSION THREAD

**Discussion by NotACat** [WinMerge/winmerge#2280] [Ideas]
I have a process that creates files that should be identical except for the date on which they are created. Would it be possible to tell WinMerge to ignore certain expected changes like that?

For example, if one folder has a set of files created on "26/02/2024" and another has the same set of files recreated on "13/03/2024" then those strings will appear in the files at corresponding places: it would be nice to be able to tell WinMerge to ignore those differences, possibly by specifying particular changes which are acceptable.

The dates come in dd/mm/yy and in dd/mm/yyyy format so it would be nice to be possible to specify more than one set of ignorable changes.

**Comment by sdottaka** [comment]
If you define Substitution filters as below, I think you will get the comparison results you expect.

![image](https://github.com/WinMerge/winmerge/assets/98126/0ec334df-e0a0-42d6-b857-c9dd558b10e8)

**Comment by NotACat** [comment]
Thanks for the suggestion but it doesn't seem to be working: could this be because I'm doing 3-way comparisons rather than 2-way?

I checked the regex with a tester (you didn't escape the "/", does that work OK in WinMerge?) so I know it matches the dates I'm seeing but the substitution is not being made. Am I missing a flag or option to make it fire up?

**Reply by sdottaka** [reply]
Indeed, it seems that the Substitutuon filter is not working well in the 3-way comparison... I'll look into it later.
Could you try using the PrediffLineFilter plugin instead, which has similar functionality?
For usage examples of PrediffLineFilter, please refer to the URL below.

https://github.com/WinMerge/winmerge/discussions/1963#discussioncomment-6621742

**Reply by NotACat** [reply]
I'll give that a shot, but I should mention that this isn't precisely what I was asking, because this will demolish all dates whereas I was just wanting to specify the particular values for which it is safe to ignore.

**Reply by sdottaka** [reply]
> because this will demolish all dates whereas I was just wanting to specify the particular values for which it is safe to ignore.

If there is a certain fixed string in the line that you want to ignore, you can specify the date and that string together as a pattern in Substitution filters, and it will not replace other dates.

Also, the problem of Substitution Filters not being applied in three-way comparisons has been fixed in the master branch. As a workaround, enable the 'Ignore blank lines' option as shown below to apply substitution filters even in 3-way comparisons.

![image](https://github.com/WinMerge/winmerge/assets/98126/1ff88645-d6af-4b14-829f-367f3c7164a6)

**Reply by NotACat** [reply]
> If there is a certain fixed string in the line that you want to ignore, you can specify the date and that string together as a pattern in Substitution filters, and it will not replace other dates.

It's not the same date across the comparison, though: can I say to replace one date in the first file, another date in the second, and yet another in the third? That doesn't seem to be available, am I looking in the wrong place?

**Reply by sdottaka** [reply]
Are you saying that you want to replace only 1/2/2024 with xx/xx/xxxx in the first file, only 2/3/2024 with xx/xx/xxxx in the second file, and only 3/4/2024 with xx/xx/xxxx in the third file? If so, unfortunately, it is not currently possible. If not, please provide specific details about the contents of the files.
